packages/belos/src/BelosPseudoBlockGmresSolMgr.hpp

```diff
@@ -366,6 +366,9 @@ namespace Belos {
     ///   call this method immediately after calling \c solve().
     bool isLOADetected() const override { return loaDetected_; }
 
+    //! Return the residual status test
+    Teuchos::RCP<StatusTestResNorm<ScalarType,MV,OP> >
+    getResidualStatusTest() const { return impConvTest_; }
```

Comment:
What do you mean by changing this one solver?
Do you mean adding the getResidualStatusTest member function?
Or, do you mean creating an ensemble version which is safe in the sense that samples may need to follow different code path (as in the case of the lucky breakdown in the GMRES)?

In both cases, my short-term plan is only to change this one solver.
However, if you mean the first possibility the change should be fast to add for all the other Belos solvers.
If you mean the second possibility, this would definitely require a consequent amount of work to identify and to treat all the occurrences of the ensemble divergence and require to add some tests.

Reply:
@kliegeois wrote:
> Do you mean adding the getResidualStatusTest member function?

Yes -- what I mean is that many Belos solvers might have a `StatusTestResNorm`, but you've only implemented this method for one solver.  Also, you've made this method return a nonconst pointer.  Do you intend users to be able to change the status test without telling the Belos solver that they did so?  If not, then please make this return `RCP<const StatusTestResNorm>`, or better yet, `const StatusTestResNorm*` (since callers really should not be keeping this pointer).